pkg/controllers/deployment/default_deployment.go

```diff
@@ -84,12 +93,54 @@ func getOAuthServerDeployment(
 	templateSpec.Volumes = append(templateSpec.Volumes, v...)
 	container.VolumeMounts = append(container.VolumeMounts, m...)
 
+	auditOpts, err := getAuditOptionsFromOpteratorConfig(&operatorConfig.Spec.ObservedConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to get audit sync data: %w", err)
+	}
+
+	container.Args[0] = strings.Replace(
+		container.Args[0],
+		"${AUDIT_OPTS}",
+		strings.Join(auditOpts, " "),
+		1,
+	)
+
+	klog.Info("xxx deployment: %+v", deployment)
+
 	return deployment, nil
 }
 
+func getAuditOptionsFromOpteratorConfig(operatorConfig *runtime.RawExtension) ([]string, error) {
+	var configDeserialized map[string]interface{}
+	oauthServerObservedConfig, err := common.UnstructuredConfigFrom(
+		operatorConfig.Raw,
+		configobservation.OAuthServerConfigPrefix,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to grab the operator config: %w", err)
+	}
+
+	if err := yaml.Unmarshal(oauthServerObservedConfig, &configDeserialized); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal the observedConfig: %v", err)
+	}
+
+	klog.Infof("xxx configDeserialized: %+v", configDeserialized)
+
+	args, _, err := unstructured.NestedStringSlice(configDeserialized, observeoauth.AuditOptionsPath...)
+	if err != nil {
+		klog.Infof("xxx ERR NestedStringSlice: %+v", err)
+		return nil, err
+	}
+
+	return args, nil
+}
```

Comment:
try to reuse this: https://github.com/openshift/cluster-authentication-operator/blob/b9de3cdbcad0771d34c011894d1322f0cf28b0b7/pkg/operator/workload/sync_openshift_oauth_apiserver.go#L253-L284

Reply:
Why?

Reply:
I hope you don't mind, when I say, that it is slightly ridiculous to parse the serverArguments, by hand into map[string][]string and then back into a string slice and then into a string. Instead of doing it immediately.

Reply:
you're free to do any optimizations, just make sure the same code is used in both places and it works